Code/Framework/AzToolsFramework/Tests/Metadata/UuidUtilTests.cpp

```diff
@@ -0,0 +1,215 @@
+
+            ON_CALL(*m_fileIOMock, Exists(_))
+                .WillByDefault(Invoke(
+                    [this](const char* filePath)
+                    {
+                        auto handle = AZ::u32(AZStd::hash<AZStd::string>{}(filePath));
+                        auto itr = m_mockFiles.find(handle);
+                        return itr != m_mockFiles.end() && itr->second.size() > 0;
+                    }));
+
+            ON_CALL(*m_fileIOMock, Read(_, _, _, _, _))
+                .WillByDefault(Invoke(
+                    [this](auto handle, void* buffer, auto, auto, AZ::u64* bytesRead)
+                    {
+                        auto itr = m_mockFiles.find(handle);
+
+                        if (itr == m_mockFiles.end())
+                        {
+                            return AZ::IO::ResultCode::Error;
+                        }
+
+                        memcpy(buffer, itr->second.c_str(), itr->second.size());
+                        *bytesRead = itr->second.size();
+                        return AZ::IO::ResultCode::Success;
+                    }));
+
+            ON_CALL(*m_fileIOMock, Write(_, _, _, _))
+                .WillByDefault(Invoke(
+                    [this](IO::HandleType fileHandle, const void* buffer, AZ::u64 size, AZ::u64* bytesWritten)
+                    {
+                        AZStd::string& file = m_mockFiles[fileHandle];
+
+                        file.resize(size);
+                        memcpy((void*)file.c_str(), buffer, size);
+                        *bytesWritten = size;
+                        return AZ::IO::ResultCode::Success;
+                    }));
+        }
+
+        void TearDown() override
+        {
+            AZ::ComponentApplicationBus::Handler::BusDisconnect();
+
+            m_jsonRegistrationContext->EnableRemoveReflection();
+            AZ::JsonSystemComponent::Reflect(m_jsonRegistrationContext.get());
+            m_jsonRegistrationContext->DisableRemoveReflection();
+
+            m_jsonRegistrationContext.reset();
+            m_serializeContext.reset();
+
+            AZ::IO::FileIOBase::SetInstance(nullptr);
+            AZ::IO::FileIOBase::SetInstance(m_priorFileIO);
+
+            UnitTest::TestRunner::Instance().ResetSuppressionSettingsToDefault();
+        }
+
+        AZStd::unique_ptr<AZ::SerializeContext> m_serializeContext;
+        AZStd::unique_ptr<AZ::JsonRegistrationContext> m_jsonRegistrationContext;
+        AZStd::unique_ptr<testing::NiceMock<AZ::IO::MockFileIOBase>> m_fileIOMock;
+        AZ::IO::FileIOBase* m_priorFileIO = nullptr;
+        AZStd::unordered_map<AZ::IO::HandleType, AZStd::string> m_mockFiles;
+        AzToolsFramework::MetadataManager m_manager;
+        AzToolsFramework::UuidUtilComponent m_uuidUtil;
+    };
+
+    TEST_F(UuidUtilTests, CreateSourceUuid_Random_ReturnsRandomUuid)
+    {
+        auto* utilInterface = AZ::Interface<AzToolsFramework::IUuidUtil>::Get();
+
+        ASSERT_TRUE(utilInterface);
```

Comment:
this happens in every test - consider moving it to Setup() and making it a member

Reply:
Will do.